### PR TITLE
integration-tests: fee too low/too high

### PR DESCRIPTION
**Pull request by tynes**
**Description**
Enforce fees in the integration tests so that
tests can be written to cover the fee rejection
logic. This was previously unit tested in geth
without end to end test coverage. Projects
depend on the error strings returned, so it is
important to have test coverage over them.

Previously the owner of the gas price oracle was set as
the same wallet used in the `l2Wallet` in the
integration tests. This would allow it to send
0 gas price transactions, as the sequencer will not
charge for transactions from the owner of the gpo.
The sequencer also updates the cached values
from the contract whenever the owner of the gpo
sends a transaction, which results in extra
logs and makes the test environment less realistic.

Updates the integration tests to work when fees
are enforced by the sequencer. This allows for
integration test coverage of the cases where transactions
are rejected when fees are too low or too high.

### CODE REVIEW

**Comment by tynes** [comment]
Enforcing fees may have downstream effects of impacting users that relied on gas price 0 transactions. There isn't really a good way to test these cases in the integration tests without turning on fee enforcement. Users can unset that config at runtime if they so choose

File: integration-tests/test/rpc.spec.ts [hunk 1]
```diff
@@ -137,6 +137,43 @@ describe('Basic RPC tests', () => {
         'gas required exceeds allowance'
       )
     })
+
+    it('should reject a transaction with too low of a fee', async () => {
+      if (isLiveNetwork()) {
+        console.log('Skipping too low of a fee test on live network')
```
Comment (mslipper) [inline integration-tests/test/rpc.spec.ts:143]: Shouldn't we `return` here or call `this.skip()`?

File: integration-tests/test/rpc.spec.ts [hunk 2]
```diff
@@ -137,6 +137,43 @@ describe('Basic RPC tests', () => {
...
+      await expect(env.l2Wallet.sendTransaction(tx)).to.be.rejectedWith(
+        `gas price too low: 1 wei, use at least tx.gasPrice = 1000 wei`
+      )
+      // Reset the gas price to its original price
+      await env.gasPriceOracle.setGasPrice(gasPrice)
+    })
+
+    it('should reject a transaction with too high of a fee', async () => {
+      if (isLiveNetwork()) {
+        console.log('Skpping too high of a fee test on live network')
```
Comment (mslipper) [inline integration-tests/test/rpc.spec.ts:163]: Ditto here - `return` or `skip`?

**Review by mslipper — APPROVED** [review]

File: integration-tests/test/shared/utils.ts [hunk 3]
```diff
@@ -47,6 +47,10 @@ const env = cleanEnv(process.env, {
   ADDRESS_MANAGER: str({
     default: '0x5FbDB2315678afecb367f032d93F642f64180aa3',
   }),
+  GAS_PRICE_ORACLE_PRIVATE_KEy: str({
```
Comment (tynes) [inline integration-tests/test/shared/utils.ts:50]: typo here, lowercase `y`

**Comment by tynes** [comment]
We have gotten the go ahead with synthetix to merge this PR

**Pull request merged**